lib/network.go

```diff
@@ -2285,6 +2285,8 @@ func DecodeHeaderVersion0(rr io.Reader) (*MsgDeSoHeader, error) {
 			return nil, errors.Wrapf(err, "MsgDeSoHeader.FromBytes: Problem decoding Height")
 		}
 		retHeader.Height = uint64(binary.LittleEndian.Uint32(scratchBytes[:]))
+		// TODO: Is this okay? Feels like it greatly simplifies things...
+		retHeader.ProposedInView = retHeader.Height
```

Comment:
See comment above. IMO we don't want to decode an artificial view directly into the PoW blocks. I would bias towards encapsulating all of the logic to validate the PoW -> PoS transition into ProcessBlockPoS, rather than distributing the logic here for a slightly simpler ProcessBlockPoS